packages/react/src/date-range-picker/utils.ts

```diff
@@ -96,3 +96,18 @@ export function getToDateInputButtonAriaLabel(
 	if (!parsed) return 'Choose end date';
 	return `Change end date, ${formatHumanReadableDate(parsed)}`;
 }
+
+/**
+ * Takes a date string and checks if it's an ISOString.
+ */
+const isISOString = (value: string) => {
+	return /\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)/.test(
+		value
+	);
+};
+
+/**
+ * Takes a string and converts ISO date strings to a Date object, otherwise converts the string to undefined.
+ */
+export const normaliseDateString = (date: string) =>
+	isISOString(date) ? new Date(date) : undefined;
```

Comment:
We shouldn't duplicate these. date-range-picker happily imports from other date-picker utils

Reply:
Oh, you're not normalising the DatePicker on L202. But I think your regex allows for invalid dates and then normalising will result in `Invalid Date`. Should we be using `parseISO` from date-fns or suggestions here https://stackoverflow.com/questions/52869695/check-if-a-date-string-is-in-iso-and-utc-format?

Reply:
Yeah ok, I can do that. FWIW I can't see why these are two separate components and feel they should be more like `Combobox` and `Button` with a base plus variants. Now we've got cross-contamination of components and discoverability is harder for consumers and devs. 

- [ ] export isISOString from date-picker utils

Reply:
>But I think your regex allows for invalid dates and then normalising will result in Invalid Date. Should we be using parseISO from date-fns or suggestions

We could use `parseISO`, sure. What dates could be invalid, though?

Reply:
Your regex allows for stuff like `'2024-00-05T01:56:20.258Z'` or `'2024-00-39T01:56:20.258Z'`